pkg/util/log/log.go

```diff
@@ -123,42 +122,38 @@ func (sw *DatadogLogger) scrub(s string) string {
 // info, warn, error, critical and off, it requires a new seelog logger because
 // an existing one cannot be updated
 func ChangeLogLevel(li seelog.LoggerInterface, level string) error {
-	l := logger.Load()
-	if l == nil {
-		return errors.New("cannot change loglevel: logger not initialized")
-	}
-
-	l.l.Lock()
-	defer l.l.Unlock()
-
-	if l.inner == nil {
-		return errors.New("cannot change loglevel: logger is initialized however logger.inner is nil")
-	}
-
-	err := l.changeLogLevel(level)
-	if err != nil {
+	if err := logger.changeLogLevel(level); err != nil {
 		return err
 	}
 
 	// See detailed explanation in SetupLogger(...)
-	err = li.SetAdditionalStackDepth(defaultStackDepth)
-	if err != nil {
+	if err := li.SetAdditionalStackDepth(defaultStackDepth); err != nil {
 		return err
 	}
 
-	l.replaceInnerLogger(li)
+	logger.replaceInnerLogger(li)
 	return nil
 
 	// need to return something, just set to Info (expected default)
 }
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏One note about this change, it means that now `changeLogLevel` and `replaceInnerLogger` are not done atomically (lock twice, not once)

Reply:
yes that's a good point, we may want to finally put lock logic to exported function, even if that may need a bit to be careful on which function we do that cause of other exported function that call each other.

We way loose some perf but it will be more safe to move lock function in a upper level in call hierarchy I think it is worth to change that, What do you think about that ?